tests/unittests/tests/01-run.py

```diff
@@ -7,13 +7,17 @@
 # directory for more details.
 
 import os, signal, sys, subprocess
-from pexpect import spawn, TIMEOUT, EOF
+import pexpect
 
 DEFAULT_TIMEOUT = 5
 
 def main():
     env = os.environ.copy()
-    child = spawn("make term", env=env, timeout=DEFAULT_TIMEOUT)
+    if sys.version_info < (3,):
```

Comment:
you can use `encoding="utf-8"`. Then you must make use of unicode strings instead of raw strings here: `r"OK \([0-9]+ tests\)"` => `u"OK \([0-9]+ tests\)"`.

```
~/r/R/t/unittests > git diff
diff --git a/tests/unittests/tests/01-run.py b/tests/unittests/tests/01-run.py
index e8b1551..e5a298c 100755
--- a/tests/unittests/tests/01-run.py
+++ b/tests/unittests/tests/01-run.py
@@ -13,10 +13,7 @@ DEFAULT_TIMEOUT = 5

 def main():
     env = os.environ.copy()
-    if sys.version_info < (3,):
-        child = pexpect.spawn("make term", env=env, timeout=DEFAULT_TIMEOUT)
-    else:
-        child = pexpect.spawnu("make term", env=env, timeout=DEFAULT_TIMEOUT)
+    child = pexpect.spawn("make term", env=env, timeout=DEFAULT_TIMEOUT, encoding="utf-8")

     child.logfile = sys.stdout

@@ -27,7 +24,7 @@ def main():
         # make reset yields error on some boards even if successful
         pass
     try:
-        child.expect(r"OK \([0-9]+ tests\)")
+        child.expect(u"OK \([0-9]+ tests\)")
     except pexpect.TIMEOUT:
         print("There where errors in the unittests")
         return 1
```

IMO it is much cleaner without checking for the python version. Tested the above diff with python2.7.11 and python3.5.1 explicitely.